sdk/storage/inc/shares/share_options.hpp

```diff
@@ -202,4 +221,675 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
     Azure::Core::Context Context;
   };
 
+  struct CreateDirectoryOptions
+  {
+    /**
+     * @brief Context for cancelling long running operations.
+     */
+    Azure::Core::Context Context;
+
+    /**
+     * @brief A name-value pair to associate with a directory object.
+     */
+    std::map<std::string, std::string> Metadata;
+
+    /**
+     * @brief This permission is the security descriptor for the directory specified in the Security
+     * Descriptor Definition Language (SDDL). If not specified, 'inherit' is used.
+     */
+    Azure::Core::Nullable<std::string> DirectoryPermission;
+
+    /**
+     * @brief SMB properties to set for the directory.
+     */
+    FileShareSmbProperties SmbProperties;
+  };
+
+  struct DeleteDirectoryOptions
+  {
+    /**
+     * @brief Context for cancelling long running operations.
+     */
+    Azure::Core::Context Context;
+  };
+
+  struct GetDirectoryPropertiesOptions
+  {
+    /**
+     * @brief Context for cancelling long running operations.
+     */
+    Azure::Core::Context Context;
+
+    /**
+     * @brief The snapshot parameter is an opaque DateTime value that, when present, specifies the
+     * share snapshot to query.
+     */
+    Azure::Core::Nullable<std::string> ShareSnapshot;
```

Comment:
Why is it here? Shouldn't we have `WithSnapshot` for share client, directory client and file client?

Reply:
ditto for others